cogs/moderation.py

```diff
@@ -1,10 +1,33 @@
 import asyncio
+from collections import Counter
+from cogs.utils.embedHandler import failure, info, success
 
 import discord
-from cogs.utils.embedHandler import failure, info, success
 from discord.ext import commands
 
-
+# Converters
+
+def can_execute_action(ctx, user, target):
+    return user.id == ctx.bot.owner_id or \
+           user == ctx.guild.owner or \
+           user.top_role > target.top_role
+
+class MemberNotFound(Exception):
+    pass
+
+async def resolve_member(guild, member_id):
+    member = guild.get_member(member_id)
+    if member is None:
+        if guild.chunked:
+            raise MemberNotFound()
+        try:
+            member = await guild.fetch_member(member_id)
```

Comment:
Why are we fetching members only, we should be able to ban users which aren't necessarily members too (meaning they aren't on the server). [`fetch_user`](https://discordpy.readthedocs.io/en/latest/api.html?highlight=fetch#discord.Client.fetch_user) will probably be a better alternative.

Note that you also need to change all occurrences of `Member` and replace it with `User` in messages, docstrings and variable names.

Reply:
I don't know Much, pls guide me abt this.

Reply:
i replaced member with user